app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt

```diff
@@ -289,6 +292,13 @@ class BrowserFragment : Fragment(), BackHandler {
                 Navigation.findNavController(view!!).navigate(directions)
                 (activity as HomeActivity).browsingModeManager.mode = BrowsingModeManager.Mode.Normal
             }
+            ToolbarMenu.Item.OpenInFenix -> {
+                val intent = Intent(context, IntentReceiverActivity::class.java)
+                val session = context!!.components.core.sessionManager.findSessionById(sessionId!!)
+                intent.action = Intent.ACTION_VIEW
+                intent.data = Uri.parse(session?.url)
+                startActivity(intent)
+            }
```

Comment:
We can file a follow-up: In theory you can re-use the same `Session`. This will avoid reloading the page and the tab will keep all its history and so on. All you need to do is to clear the `CustomTabConfig` and then select the `Session` in the `SessionManager`.

Reply:
I tried implementing this, but am getting a `displayAlreadyAcquired` crash when doing so. I believe we need to [fix this bug](https://github.com/mozilla-mobile/fenix/issues/897) before we can implement this

Reply:
Follow up issue here: https://github.com/mozilla-mobile/fenix/issues/966